src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CollectionPropertiesShouldBeReadOnly.cs

```diff
@@ -56,31 +56,26 @@ public override void Initialize(AnalysisContext analysisContext)
             analysisContext.RegisterCompilationStartAction(
                 (context) =>
                 {
-                    INamedTypeSymbol? iCollectionType = context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemCollectionsICollection);
-                    INamedTypeSymbol? genericICollectionType = context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemCollectionsGenericICollection1);
-                    INamedTypeSymbol? arrayType = context.Compilation.GetSpecialType(SpecialType.System_Array);
-                    INamedTypeSymbol? dataMemberAttribute = context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeSerializationDataMemberAttribute);
-                    ImmutableHashSet<INamedTypeSymbol> immutableInterfaces = GetIImmutableInterfaces(context.Compilation);
-
-                    if (iCollectionType == null ||
-                        genericICollectionType == null ||
-                        arrayType == null)
+                    var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(context.Compilation);
+                    var knownTypes = new KnownTypes(wellKnownTypeProvider);
```

Comment:
Create a wrapping type instead of adding 3 more parameters. Let me know if you prefer the other solution.